### ISSUE TITLE
Announce changes in non-empty selections on Android

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#14256]
With https://github.com/flutter/engine/pull/4587 moving the cursor (without an active selection) announces the letter the cursor has been moved over as expected.

In theory, that change should also announce cursor movements properly when changing a selection in selection mode. However, no announcement is made in that case.

The adb log shows that in this case Android in response to our `TYPE_VIEW_TEXT_SELECTION_CHANGED` event generates the following (incorrect) internal event:

```
01-24 16:09:37.328  3802  3802 I TextEventInterpreter: interpretation:  event=EVENT_TYPE_INPUT_CHANGE_INVALID reason="Unhandled selection event." textOrDescription="Hello "
```

To be correct, it should have generated something like:

```
01-24 16:11:01.458  3802  3802 I TextEventInterpreter: interpretation:  event=EVENT_TYPE_INPUT_SELECTION_MOVE_CURSOR_WITH_SELECTION reason="Selecting and toIndex == cursorPosition" textOrDescription="hello " selectedText="e"
```

It's unclear why Android doesn't do that :(

**Comment by goderbauer** [comment]
This only happens when the selection is extended backwards, e.g. 123|4| -> 12|34| doesn't work. However, |1|234 -> |12|34 works as expected.

Looks like the problem is that we always send textSelectionStart < textSelectionEnd from the framework. Even when it should be textSelectionStart > textSelectionEnd like in the first case.
